File: 1642-water-bottles/water-bottles.py
class Solution(object):
    def numWaterBottles(self, numBottles, numExchange):
        x = numBottles
        y = numBottles

        while y >= numExchange:
            z = y // numExchange
            x += z
            y = z + (y % numExchange)
        
        return x

File: 1642-water-bottles/water-bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int x = numBottles;
        int y = numBottles;

        while (y >= numExchange) {
            int z = y / numExchange;
            x += z;
            y = z + (y % numExchange);
        }

        return x;
    }
}

